src/printer-postcss.js

```diff
@@ -0,0 +1,357 @@
+      }
+
+      return group(indent(concat(parts)));
+    }
+    case "value-paren_group": {
+      const parent = path.getParentNode();
+      const isURLCall =
+        parent && parent.type === "value-func" && parent.value === "url";
+
+      if (
+        isURLCall &&
+        (n.groups.length === 1 ||
+          (n.groups.length > 0 &&
+            n.groups[0].type === "value-comma_group" &&
+            n.groups[0].groups.length > 0 &&
+            n.groups[0].groups[0].type === "value-word" &&
+            n.groups[0].groups[0].value === "data"))
+      ) {
+        return concat([
+          n.open ? path.call(print, "open") : "",
+          join(",", path.map(print, "groups")),
+          n.close ? path.call(print, "close") : ""
+        ]);
+      }
+
+      if (!n.open) {
+        return group(
+          indent(join(concat([",", line]), path.map(print, "groups")))
+        );
+      }
+
+      return group(
+        concat([
+          n.open ? path.call(print, "open") : "",
+          indent(
+            concat([
+              softline,
+              join(concat([",", line]), path.map(print, "groups"))
+            ])
+          ),
+          softline,
+          n.close ? path.call(print, "close") : ""
+        ])
+      );
+    }
+    case "value-value": {
+      return path.call(print, "group");
+    }
+    case "value-func": {
+      return concat([n.value, path.call(print, "group")]);
+    }
+    case "value-paren": {
+      if (n.raws.before !== "") {
+        return concat([line, n.value]);
+      }
+      return n.value;
+    }
+    case "value-number": {
+      return concat([n.value, n.unit]);
+    }
+    case "value-operator": {
+      return n.value;
+    }
+    case "value-word": {
+      return n.value;
+    }
+    case "value-colon": {
+      return n.value;
+    }
+    case "value-comma": {
+      return concat([n.value, " "]);
+    }
+    case "value-string": {
+      return concat([
+        n.quoted ? n.raws.quote : "",
+        n.value,
+        n.quoted ? n.raws.quote : ""
+      ]);
+    }
+    case "value-atword": {
+      return concat(["@", n.value]);
+    }
+
+    default:
+      throw new Error("unknown postcss type: " + JSON.stringify(n.type));
+  }
+}
+
+function printNodeSequence(path, options, print) {
+  const node = path.getValue();
+  const parts = [];
+  let i = 0;
+  path.map(pathChild => {
+    parts.push(pathChild.call(print));
+    if (i !== node.nodes.length - 1) {
+      if (
+        node.nodes[i + 1].type === "css-comment" &&
+        !util.hasNewline(
+          options.originalText,
+          util.locStart(node.nodes[i + 1]),
+          { backwards: true }
+        )
+      ) {
+        parts.push(" ");
+      } else {
+        parts.push(hardline);
+        if (util.isNextLineEmpty(options.originalText, pathChild.getValue())) {
+          parts.push(hardline);
+        }
+      }
+    }
+    i++;
+  }, "nodes");
+
+  return concat(parts);
+}
+
+function printValue(value) {
```

Comment:
Looks like i'm no longer making use of this function anymore, this can be removed :)